### PR TITLE
Add TruffleRuby 23.0.0

### PR DESCRIPTION
**Pull request by eregon**
* Use the new Oracle GraalVM distribution which uses the GFTC license as it is significantly faster than GraalVM CE and is free for development and production use.
* See https://medium.com/graalvm/whats-new-in-graalvm-languages-161527df3d76

### CODE REVIEW

File: scripts/functions/selector_interpreters
Comment (eregon) [inline scripts/functions/selector_interpreters:279]: These URLs are not pretty, but they are stable and should keep working forever.
We're working on getting a stable URL format for them, similar to `truffleruby+graalvm-23.0.0` in the other file.

**Comment by eregon** [comment]
There is some failure at https://github.com/rvm/rvm/actions/runs/5312825778/jobs/9617886118?pr=5359, I'll take a look

File: scripts/functions/manage/truffleruby
Comment (eregon) [reply scripts/functions/manage/truffleruby:13]: I was thinking a one-time heads up, and revisit when the next release comes out.
Could also log for a few more releases if desired. I think better to not log it for all future releases.

Comment (havenwood) [reply scripts/functions/manage/truffleruby:13]: A one-time heads up seems nice. 👍

File: scripts/functions/manage/truffleruby
Comment (havenwood) [reply scripts/functions/manage/truffleruby:12]: Since the logging is meant to be a one-time heads up, this comment only applies to the `elif` logic below.

Comment (eregon) [reply scripts/functions/manage/truffleruby:12]: For the `elif` in `scripts/functions/selector_interpreters`, that needs to compare exactly to 23.0.0, because for e.g. 23.1.0 or so it would be different URLs.

**Comment by eregon** [comment]
The CI should be fixed now (well, except macOS which fails for an [unrelated reason](https://github.com/rvm/rvm/actions/runs/5312825778/jobs/9617886385?pr=5359)).
I have also added a Dockerfile to quickly test changes in RVM without affecting the local system, this proved very valuable to fix the CI here and also to notice I forgot to add the dependency on libyaml (now added).

**Review by havenwood — APPROVED** [review]
LGTM. cc @pkuczynski

**Comment by eregon** [comment]
Could someone merge this?

BTW, GitHub shows me:
Screenshot from 2023-06-21 12-23-50

So I can't merge PRs even after an approval and all checks passed, could that be fixed?
Regarding the signed commits, I think that would be addressed by doing a rebase & merge (or squash & merge but that would lose the individual commits, seems bad if we ever have to bisect). Also not sure of the value to require signed commits, it seems significant extra setup for every contributor (and I think creating the commit on GitHub doesn't make it any safer).

**Comment by havenwood** [comment]
> So I can't merge PRs even after an approval and all checks passed, could that be fixed? Regarding the signed commits, I think that would be addressed by doing a rebase & merge (or squash & merge but that would lose the individual commits, seems bad if we ever have to bisect). Also not sure of the value to require signed commits, it seems significant extra setup for every contributor (and I think creating the commit on GitHub doesn't make it any safer).

@eregon Just to sanity check, I confirmed you for sure have write permission to rvm/rvm. Yeah, I agree the signing requirement doesn't make sense for merging your own commits, since we know we're ourselves. It makes sense for avoiding merging spoofed PRs from others based on their reputation. I had automatic commit signing set up but just swapped boxes and this reminds me I need to set it up again, since it really is a pain to do manually.

@eregon With our current repo constraints, I'd have to squash and merge to bypass branch protection too, unfortunately. As long as the signed commit setting is in place, I'm afraid squashing or individually signing is the only way to appease the branch protection.

@pkuczynski What do you think about the signed commit requirement? Is the identity confirmation we gain worth the hassle of contributors setting up automatic signing to keep commit history?

**Comment by eregon** [comment]
I signed all commits, but I still cannot merge:
Screenshot from 2023-06-22 20-13-47

TBH I feel it's quite a hassle to get a PR merged in RVM these days.
But at least @havenwood should be able to merge with commit or with rebase now.

**Comment by havenwood** [comment]
@eregon and I can't merge without a "Squash and merge" still, even with all commits signed. Let me know if you want me to go ahead and squash and I'll merge right away. I don't know what's blocking you from merging and making me squash, but it seems it's non-standard GitHub settings enabled.

@pkuczynski, could we relax "Squash" requirement for when a single PR has multiple discrete commits?

**Comment by havenwood** [comment]
@eregon we should for sure figure out why you're unable to merge. I looked at GitHub permissions and you have `write` for sure, so it must be some other branch protection configuration.

**Comment by eregon** [comment]
@havenwood  Could you squash and merge then? At least we'll have the separate commits in this PR.

**Comment by havenwood** [comment]
Congratulations on the release of TruffleRuby 23.0!

**Pull request merged**
